### DISCUSSION TITLE
Show Context with Logs

### DISCUSSION THREAD

**Discussion by lahivee** [spatie/laravel-ray#162] [Ideas]
Ray currently doesn't show the context passed to the log methods:

```
Log::error(string $message, array $context);
```

**Comment by freekmurze** [comment]
I'd accept a PR that adds this. If none comes in, I'll add it myself  in the future.

**Reply by craftyshaun** [reply]
@freekmurze is this able to be done with just alterations to laravel-ray?

I thought it would need the `ApplicationLogPayload` to be altered to add a context prop. Then you would render in the (closed source) Ray frontend with some electron code.

I'm willing to try a PR but I don't feel right making architecture decisions. I feel you created an `ApplicationLogPayload` for a good reason.

Feel free to provide any hints as we started to discuss this in the ideas section.

**Comment by patinthehat** [comment]
@craftyshaun It looks like this needs to be two PRs: one to `spatie/ray` _(updating `ApplicationLogPayload`)_ and one to `spatie/laravel-ray` to send the context to `ApplicationLogPayload`.  Are you planning on working on this feature? If not, I'll give it a go.

**Reply by craftyshaun** [reply]
@patinthehat you can try it but doesn't the Ray app need to know how to render the payload (and it's properties) to the correct HTML.

I started a idea on the spatie/ray repo discussing this.

If you can do it without front end changes  feel free to take a go at it.

It's late night here in AU now but if you open a PR or comment a Wip branch I'm happy to take a look tomorrow and collab.

**Comment by yakatz** [comment]
Did anything ever happen with this?

**Comment by bilfeldt** [comment]
Context is now passed correctly to the Ray app, but the core team needs to add display this context as that is part of the core app in a closed repo.

https://github.com/spatie/laravel-ray/pull/293
